src/main/scala/com/microsoft/hyperspace/index/IndexCollectionManager.scala

```diff
@@ -34,11 +34,13 @@ class IndexCollectionManager(
   private val conf: SQLConf = spark.sessionState.conf
 
   override def create(df: DataFrame, indexConfig: IndexConfig): Unit = {
-    val indexPath = PathResolver(spark.sessionState.conf).getIndexPath(indexConfig.indexName)
-    val dataManager = indexDataManagerFactory.create(indexPath)
+    val indexPath = PathResolver(spark.sessionState.conf, spark.sessionState.newHadoopConf())
```

Comment:
shall we create a variable for `spark.sessionState.newHadoopConf()` and reuse it? I see `newHadoopConf` three times here.

Reply:
Looks like it's used many times in this class.
We can either create one `lazy val` at the class level or create a variable at a function level if caching at class level is a concern for staleness.

Reply:
Yes, caching may be an issue as the shared spark's Hadoop configuration may change and we want to have it updated in Hyperspace too. I'll create method variables.